TAInformation/views.py

```diff
@@ -105,14 +105,96 @@ def post(self, request):
                           {"message": "insufficent permissions to create a course. Please contact "
                                       "your system administrator if you believe this is in error."})
         else:
-            newCourse = addCourse(request.POST['name'], request.POST['instructor'],
+          newCourse = addCourse(request.POST['name'], request.POST['instructor'],
                                   request.POST['meeting_time'], request.POST['semester'], request.POST['course_type'],
                                   request.POST['description'])
-            newCourse.save()
-            return render(request, "dashboard.html",
-                          {"message": "Course Created Successfully"})
-
+          newCourse.save()
+          return render(request, "dashboard.html",
+                        {"message": "Course Created Successfully"})
+        
+        
+# For create account web pages
+class CreateUser(View):
+    # Precondition: User is logged in and data is stored in session
+    # Postcondition: Page with Create Course form is displayed
+    # Side Effects: None
+    def get(self, request):
+        return render(request, "create_user.html", {})
 
+    # Precondition: User has entered data in all form fields in proper format
+    # Postcondition: Creates new user, if data entered validates successfully and user doesn’t already exist.
+    # Side Effects: Message indicating result is displayed at the bottom of the “Create Courses” form
+    def post(self, request):
+        # Extract data from form
+        match request.session['role']:
+            case AccountType.ADMIN.value:
+                current_user = UserAdmin(
+                    int(request.session['user_id']),
+                    "",
+                    "",
+                    request.session['email'],
+                    "",
+                    ""
+                )
+            case AccountType.INSTRUCTOR.value:
+                current_user = Instructor(
+                    int(request.session['user_id']),
+                    "",
+                    "",
+                    request.session['email'],
+                    "",
+                    ""
+                )
+            case AccountType.TA.value:
+                current_user = TA(
+                    int(request.session['user_id']),
+                    "",
+                    "",
+                    request.session['email'],
+                    "",
+                    ""
+                )
+            case _:
+                current_user: BaseUser
+
+        result_dict = {}
+        the_id = -1 if request.POST.get('user_id') == "" else int(request.POST.get('user_id'))
+        match int(request.POST.get('role')):
```

Comment:
What's the purpose of instantiating the class object, just to save its data into the database? Is it possible to just get the data from the form and put it into the database? Then you don't even need to use any case statements.

Reply:
I could try using a User model object instead and change the create_user and validators to work for a model instead.